pkg/operator/controller/status/controller.go

```diff
@@ -297,21 +298,25 @@ func (r *reconciler) computeOperatorStatusVersions(oldVersions []configv1.Operan
 }
 
 // checkAllIngressesAvailable checks if all the ingress controllers are available.
-func checkAllIngressesAvailable(ingresses []operatorv1.IngressController) bool {
+func checkAllIngressesAvailable(ingresses []operatorv1.IngressController) error {
+	errs := make([]error, 0, len(ingresses))
 	for _, ing := range ingresses {
-		available := false
+		found := false
 		for _, c := range ing.Status.Conditions {
-			if c.Type == operatorv1.IngressControllerAvailableConditionType && c.Status == operatorv1.ConditionTrue {
-				available = true
+			if c.Type == operatorv1.IngressControllerAvailableConditionType {
+				if c.Status != operatorv1.ConditionTrue {
+					errs = append(errs, fmt.Errorf("ingress controller %s in namespace %s is Available=%s: %s: %s", ing.Name, ing.Namespace, c.Status, c.Reason, c.Message))
```

Comment:
```suggestion
					errs = append(errs, fmt.Errorf("ingress controller %s in namespace %s is not Available=%s: %s: %s", ing.Name, ing.Namespace, c.Status, c.Reason, c.Message))
```

Reply:
Isn't `c.Status` dynamic here -- "not" doesn't always hold true.

Reply:
Or `"ingress controller test-1 is Available=Unknown`